sync/class.jetpack-sync-dashboard.php

```diff
@@ -0,0 +1,163 @@
+				'action' => __( 'Reindex Posts', 'jetpack' ),
+				'status' => __( 'Posts indexed.', 'jetpack' ),
+			),
+			'ERROR'       => array(
+				'action' => __( 'Refresh Status', 'jetpack' ),
+				'status' => __( 'Status unknown.', 'jetpack' ),
+			),
+			'ERROR:LARGE' => array(
+				'action' => __( 'Refresh Status', 'jetpack' ),
+				'status' => __( 'This site is too large, please contact Jetpack support to sync.', 'jetpack' ),
+			),
+		);
+		$initial_queue_status     = json_encode( $this->queue_status() );
+		$initial_full_sync_status = json_encode( $this->full_sync_status() );
+
+		wp_localize_script( 'jetpack_sync_reindex_control', 'sync_dashboard', array(
+			'possible_status'  => $strings,
+			'queue_status'     => $initial_queue_status,
+			'full_sync_status' => $initial_full_sync_status
+		) );
+	}
+
+	function page_render() {
+		$this->dashboard_ui();
+	}
+
+	function init() {
+		add_action( 'wp_ajax_jetpack-sync-queue-status', array( $this, 'ajax_queue_status' ) );
+		add_action( 'wp_ajax_jetpack-sync-reset-queue', array( $this, 'ajax_reset_queue' ) );
+		add_action( 'wp_ajax_jetpack-sync-unlock-queue', array( $this, 'ajax_unlock_queue' ) );
+		add_action( 'wp_ajax_jetpack-sync-begin-full-sync', array( $this, 'ajax_begin_full_sync' ) );
+		add_action( 'wp_ajax_jetpack-sync-cancel-full-sync', array( $this, 'ajax_cancel_full_sync' ) );
+		add_action( 'wp_ajax_jetpack-sync-full-sync-status', array( $this, 'ajax_full_sync_status' ) );
+
+	}
+
+	// returns size of queue and age of oldest item (aka lag)
+	function ajax_queue_status() {
+		$response = json_encode( $this->queue_status() );
+		echo $response;
+		exit;
+	}
+
+	function ajax_reset_queue() {
+		Jetpack_Sync_Client::getInstance()->reset_sync_queue();
+		echo json_encode( array( 'success' => true ) );
+		exit;
+	}
+
+	function ajax_unlock_queue() {
+		Jetpack_Sync_Client::getInstance()->get_sync_queue()->force_checkin();
+		echo json_encode( array( 'success' => true ) );
+		exit;
+	}
+
+	function ajax_begin_full_sync() {
+		Jetpack_Sync_Client::getInstance()->get_full_sync_client()->start();
+		$this->ajax_full_sync_status();
+	}
+
+	function ajax_cancel_full_sync() {
+		// TODO	
+	}
+
+	function ajax_full_sync_status() {
+		$response = json_encode( $this->full_sync_status() );
+		echo $response;
+		exit;
+	}
+
+	function queue_status() {
+		$client = Jetpack_Sync_Client::getInstance();
+		$queue  = $client->get_sync_queue();
+
+		return array(
+			'size' => $queue->size(),
+			'lag'  => $queue->lag()
+		);
+	}
+
+	function full_sync_status() {
+		$client = Jetpack_Sync_Client::getInstance();
+
+		return $client->get_full_sync_client()->get_complete_status();
+	}
+
+	function dashboard_ui() {
+		wp_enqueue_script( 'jetpack_sync_reindex_control' );
+		?>
+		<div class="wrapper">
+			<div class="page-content">
+				<div id="sync_status">
+					Sync status:
+				</div>
+				<p><strong>Warning: Clicking either of these buttons can get you out of sync!</strong></p>
```

Comment:
We intend to completely replace this UI before release with one written in React, so will add translation code then.